File: computer_architecture/cs_primer/machine_code/sum.c
#include <stdio.h>
#include <stdlib.h>

int sum(int *arr, int n) {
    int total = 0;
    
    for (int i = 0; i < n; i++)
        total += arr[i];
    
    return total;
}

int main() {
    int arr[] = {5, 20, 3};

    int total = sum(arr, sizeof(arr) / sizeof(int));
    printf("Total: %d\n", total);
    exit(0);
}

File: go/5/examples/sum.go
package main

import "fmt"


func main() {

	fmt.Println(sum(1,2,3,45,6,7,78,8,10))

}

func sum(vals ...int) int {
	total := 0
	for _, val := range vals {
		total += val
	}

	return total
}